packages/components/src/templates/next/components/complex/ImageGallery/utils/getPreviewIndices.ts

```diff
@@ -0,0 +1,30 @@
+interface GetPreviewIndicesProps {
+  numberOfImages: number
+  currentIndex: number
+  maxPreviewImages: 3 | 5
+}
+
+export const getPreviewIndices = ({
+  numberOfImages,
+  currentIndex,
+  maxPreviewImages,
+}: GetPreviewIndicesProps) => {
+  // If there are less than or equal to max images, show all of them
+  if (numberOfImages <= maxPreviewImages) {
+    return Array.from({ length: numberOfImages }, (_, i) => i)
+  }
+
+  const numberOfImagesBeforeCurrent = Math.floor(maxPreviewImages / 2)
```

Comment:
not sure i understand this line here. if i have an image array [0, 1, 2, 3, 4], and i'm at position 4, the number of images before current should be 4 isn't it? similarly if i'm at position 1, it's should be 0.

if the direction is descending, then we can just flip the logic so i'm not entirely sure i understand why it's `maxPreviewImages / 2`